### PR TITLE
docs/why conventional commits decision post

### LINKED ISSUE #90: Add a decision post on using Conventional Commits

### PR DESCRIPTION
**Pull request by signekb**
## Description

- This PR adds a post on why we follow the Conventional Commits conventions. 
- As always, feedback/suggestions/comments are more than welcome. I'm a bit uncertain on the difference between the sections "Context and problem statement" and "Decision drivers"

## Related Issues

Closes seedcase-project/team#90

See also Issues #...

## Testing

- [X] No, not needed (give a reason below)

Documentation

## Reviewer Focus

This PR requires an in-depth review.

## Checklist

For general documentation:

- [ ] Spell-check
    - [ ] US
    - [ ] UK
- [X] Did the page(s) preview correctly on your machine without breaking
- [ ] New category words (keywords) (if any) added to the code snippet file

### CODE REVIEW

File: why-conventional-commits.qmd [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+    irrelevant commits, like formatting).
+:::
+
+::: column
+#### Drawbacks
+
+-   It can be too verbose for some projects.
+-   Usage more prevalent in Angular-related projects.
+-   Developed in a Typescript context
+-   No emojis which can make commit messages easier to scan and more
```
Comment (K-Beicher) [inline why-conventional-commits.qmd:142]: It isn't completely clear to me if the emojis makes it easier or more difficult to scan the message :-)

File: why-conventional-commits.qmd [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+    visually appealing.
+:::
+:::
+
+### Conventional commits
+
+Conventional commits are inspired by the Angular convention described
+above, but is more lightweight (how?). This convention also provides a
```
Comment (K-Beicher) [inline why-conventional-commits.qmd:150]: Did you mean to leave the (how?) in here?

**Review by K-Beicher — DISMISSED** [review]
Wonderful.  I was wondering if an example of a commit could be described in the blurb at the beginning of the Options, and then written out in each of the conventions, that would help me a lot to understand the differences between the first two in particular.

File: why-conventional-commits.qmd (hunk 2 above)
Comment (signekb) [reply why-conventional-commits.qmd:150]: Haha, nope!

**Comment by signekb** [comment]
@K-Beicher Thanks for your feedback :) Yeah, they are very similar - bascially, bc the Conventional Commits convention is based on Angular. I talked to @lwjohnst86 about this and I'll remove Angular. 

I'll convert this PR into a draft again and rework it 👍

File: why-conventional-commits.qmd (hunk 1 above)
Comment (signekb) [reply why-conventional-commits.qmd]: I have slightly rewritten this. I hope that it's clearer now :)

File: why-conventional-commits.qmd (hunk 2 above)
Comment (signekb) [reply why-conventional-commits.qmd]: Thanks for the catch. I have removed it ;)

**Comment by signekb** [comment]
It's now ready for review again 🚀

File: why-conventional-commits.qmd [hunk 3]
```diff
@@ -0,0 +1,249 @@
+---
+title: "Why conventional commits (with optional emojis)"
+description: "Our reasons for using conventional commits (with optional emojis following the Gitmoji convention) across the Seedcase projects."
+date: "2024-05-14"
+categories:
```
Comment (lwjohnst86) [inline why-conventional-commits.qmd:5]:
```suggestion
categories:
- git
- commit messages
- communication
- collaboration
- teamwork
- workflow
```

File: why-conventional-commits.qmd [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+> messages in Seedcase projects?
+
+## Decision drivers
+
+::: content-hidden
+List some reasons for why we need to make this decision and what things
+have arisen that impact work.
+:::
+
+In the Seedcase projects, we emphasise the open-science philosophies of
```
Comment (lwjohnst86) [inline why-conventional-commits.qmd:47]:
```suggestion
In Seedcase projects, we emphasise the open-source philosophies of
```

File: why-conventional-commits.qmd [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+
+### Conventional commits
+
+The [Conventional Commits](https://www.conventionalcommits.org/)
+convention builds on the [Angular
+convention](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#-commit-message-guidelines),
+but was generalised to go beyond Typescript. It's widely adopted due to
+its compatibility with semantic versioning and automated release
+processes. Moreover, it's commonly used in various software development
+communities, including the open science communities. Its structured
```
Comment (lwjohnst86) [inline why-conventional-commits.qmd:102]:
```suggestion
communities, including the open source communities. Its structured
```

Open source, not science right? Commit message conventions are not well used within open science unfortunately... Version control isn't as common either!

File: why-conventional-commits.qmd [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+format aligns well with the organized and systematic approach favoured
+in open-science projects. Additionally, its is compatible with automated
```
Comment (lwjohnst86) [inline why-conventional-commits.qmd:104]:
```suggestion
in open-source projects. It is also compatible with automated
```

File: why-conventional-commits.qmd [hunk 7]
```diff
@@ -0,0 +1,249 @@
+---
+title: "Why conventional commits (with optional emojis)"
+description: "Our reasons for using conventional commits (with optional emojis following the Gitmoji convention) across the Seedcase projects."
```
Comment (lwjohnst86) [inline why-conventional-commits.qmd:3]:
```suggestion
description: "Our reasons for using conventional commits (with optional emojis following the Gitmoji convention) across Seedcase projects."
```

File: why-conventional-commits.qmd [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+that eases the process of writing commit messages in the correct format.
+
+::: columns
+::: column
+#### Benefits
+
+-   Provides a clear structure for commit messages.
+-   Has an easy-to-use VS Code extension.
+-   Can be used to generate changelogs.
+-   Widespread adoption in various software development and open science
```
Comment (lwjohnst86) [inline why-conventional-commits.qmd:148]:
```suggestion
-   Widespread adoption in various software development and open source
```

File: why-conventional-commits.qmd [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+::: columns
+::: column
+#### Benefits
+
+-   Provides a clear structure for commit messages.
+-   Has an easy-to-use VS Code extension.
+-   The use of emojis can make commit messages more visually appealing
+    and easier to scan to identify the purpose or intention.
+-   Gitmoji is an open-source project which fits well with the open
+    science philosophy of the Seedcase projects.
```
Comment (lwjohnst86) [inline why-conventional-commits.qmd:190]:
```suggestion
    source philosophy of the Seedcase Project.
```

File: why-conventional-commits.qmd [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+:::
+
+## Decision outcome
+
+::: content-hidden
+What decision was made, use the form "We decided on CHOICE because of
+REASONS."
+:::
+
+When deciding which commit message convention to follow in the Seedcase
```
Comment (lwjohnst86) [inline why-conventional-commits.qmd:211]:
```suggestion
When deciding which commit message convention to follow in all Seedcase
```

**Review by lwjohnst86 — CHANGES_REQUESTED** [review]
Wonderful 😁 Some small edits

File: why-conventional-commits.qmd (hunk 5 above)
Comment (signekb) [reply why-conventional-commits.qmd:102]: Ah, yes! Thanks for the catch!

File: why-conventional-commits.qmd [hunk 11]
```diff
@@ -0,0 +1,249 @@
...
+
+::: columns
+::: column
+#### Benefits
+
+-   Provides a clear structure for commit messages.
+-   Has an easy-to-use VS Code extension.
+-   The use of emojis can make commit messages more visually appealing
+    and easier to scan to identify the purpose or intention.
+-   Gitmoji is an open-source project which fits well with the open
```
Comment (signekb) [inline why-conventional-commits.qmd:189]:
```suggestion
-   Gitmoji is an open-source project which fits well with the open-source
```

File: why-conventional-commits.qmd [hunk 12]
```diff
@@ -0,0 +1,261 @@
...
+- communication
+- collaboration
+- teamwork
+- workflow
+- git
+- commit messages
+- communication
+- collaboration
+- teamwork
+- workflow
```
Comment (signekb) [inline why-conventional-commits.qmd:17]:
```suggestion
```

**Comment by signekb** [comment]
@lwjohnst86 I think that I have reviewed and implemented your requested changes, but it still seems there's one change requested? :)

**Review by lwjohnst86 — APPROVED** [review]

**Pull request merged**